pkg/jaeger/store/store.go

```diff
@@ -45,7 +49,17 @@ func (p *Store) SpanWriter() spanstore.Writer {
 }
 
 func (p *Store) StreamingSpanWriter() spanstore.Writer {
-	return p
+	return streamingSpanWriter{p}
+}
+
+func (s streamingSpanWriter) WriteSpan(ctx context.Context, span *model.Span) error {
+	go func() {
```

Comment:
Should we add some back-preasure mechanism? Something like putting it behind a queue with a pool of goroutines that call the `WriteSpan`?

Reply:
Yes backpressure is key here. Why isn't it enough to simply turn off async-acks for the streaming stuff

Reply:
Sure, lets add a back pressure based on the defaults from https://github.com/jaegertracing/jaeger/blob/c1bb2946e670129314bf88d9730d8ed7566766d4/plugin/storage/grpc/shared/streaming_writer.go#L32.

Reply:
Looks like I missed the point that the existing batching implementation already has the [back-pressure mechanism](https://github.com/timescale/promscale/blob/efec5ae94de49b1afcca4d3abeff13b8b14b2c59/pkg/pgmodel/ingestor/trace/trace_dispatcher.go#L63) and I can completely skip creating new go routine and turn on async ack as mentioned by @cevian .